<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>645</width>
    <height>583</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hopfield Network</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="3,1,3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="3,2">
      <item>
       <widget class="QGroupBox" name="groupWorkspace">
        <property name="title">
         <string>Workspace</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6" stretch="0,1,0">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,10,1">
           <item>
            <widget class="QPushButton" name="btnWorkspaceLeft">
             <property name="minimumSize">
              <size>
               <width>50</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>&lt;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lblWorkspace">
             <property name="text">
              <string>0/0</string>
             </property>
             <property name="scaledContents">
              <bool>false</bool>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnWorkspaceRight">
             <property name="minimumSize">
              <size>
               <width>50</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QWidget" name="mplWorkspace" native="true">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="mplWorkspaceLayout"/>
          </widget>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QPushButton" name="btnNewPattern">
             <property name="text">
              <string>New pattern</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="btnDeletePattern">
             <property name="text">
              <string>Delete pattern</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="btnClearPattern">
             <property name="text">
              <string>Clear pattern</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="btnSetAsInput">
             <property name="text">
              <string>Set as input →</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupSettings">
        <property name="title">
         <string>Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QFormLayout" name="formLayout">
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <property name="labelAlignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="xSizeLabel">
             <property name="text">
              <string>X Size</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="spnXSize">
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="ySizeLabel">
             <property name="text">
              <string>Y Size</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="spnYSize">
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="distortionLabel">
             <property name="text">
              <string>Distortion %</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="spnDistortion">
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="buttonApplySettings">
             <property name="text">
              <string>Apply (resets everything)</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="animationLabel">
             <property name="text">
              <string>Animation</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QCheckBox" name="chkAnimation">
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="animationSpeedLabel">
             <property name="text">
              <string>Animation speed</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QSlider" name="sldAnimationSpeed">
             <property name="maximum">
              <number>10</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
             <property name="sliderPosition">
              <number>5</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_4" stretch="1,1,1,6">
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnLearn">
        <property name="text">
         <string>Learn →</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnLearnAll">
        <property name="text">
         <string>Learn all →→</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3" stretch="8,0,7">
      <item>
       <widget class="QGroupBox" name="groupLearned">
        <property name="title">
         <string>Learned patterns</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_7" stretch="0,1,0">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,5,1">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <widget class="QPushButton" name="btnLearnedLeft">
             <property name="minimumSize">
              <size>
               <width>50</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>&lt;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lblLearned">
             <property name="text">
              <string>0/0</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnLearnedRight">
             <property name="minimumSize">
              <size>
               <width>50</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QWidget" name="mplLearned" native="true">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="mplLearnedLayout"/>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QPushButton" name="btnUnlearn">
             <property name="text">
              <string>x Unlearn x</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnDistort">
        <property name="text">
         <string>↓ Distort ↓</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupIO">
        <property name="title">
         <string>Input/output</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_8" stretch="1,0">
         <item>
          <widget class="QWidget" name="mplIO" native="true">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="mplIOLayout"/>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="btnSolve">
             <property name="text">
              <string>Solve!</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnSaveToWorkspace">
             <property name="text">
              <string>Save to workspace</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>645</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuWorkspace">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_text"/>
    <addaction name="actionLoad_image"/>
    <addaction name="separator"/>
    <addaction name="actionSave_workspace"/>
    <addaction name="actionSave_output"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuNetwork">
    <property name="title">
     <string>Network</string>
    </property>
    <addaction name="actionUnlearn_all"/>
    <addaction name="actionWeight_matrix"/>
   </widget>
   <addaction name="menuWorkspace"/>
   <addaction name="menuNetwork"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_text">
   <property name="text">
    <string>Load pattern(s) from text...</string>
   </property>
  </action>
  <action name="actionLoad_image">
   <property name="text">
    <string>Load pattern from image...</string>
   </property>
  </action>
  <action name="actionSave_Workspace_Text">
   <property name="text">
    <string>Save Workspace [Text]</string>
   </property>
  </action>
  <action name="actionSave_Output_Text">
   <property name="text">
    <string>Save Output [Text]</string>
   </property>
  </action>
  <action name="actionSave_Workspace">
   <property name="text">
    <string>Workspace to text file</string>
   </property>
  </action>
  <action name="actionClear_workspace">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionSave_workspace">
   <property name="text">
    <string>Save workspace...</string>
   </property>
  </action>
  <action name="actionSave_output">
   <property name="text">
    <string>Save output...</string>
   </property>
  </action>
  <action name="actionShow_weight_matrix">
   <property name="text">
    <string>Show weight matrix</string>
   </property>
  </action>
  <action name="actionUnlearn_all">
   <property name="text">
    <string>Unlearn all patterns</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionWeight_matrix">
   <property name="text">
    <string>Show weight matrix</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
